data-prepper-plugins/encryption-plugin/src/main/java/org/opensearch/dataprepper/plugins/encryption/KmsEncryptionEngineConfiguration.java

```diff
@@ -0,0 +1,162 @@
+import org.opensearch.dataprepper.aws.api.AwsContext;
+import org.opensearch.dataprepper.aws.api.AwsCredentialsConfig;
+import org.opensearch.dataprepper.aws.api.AwsCredentialsOptions;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+
+@JsonTypeName(KmsEncryptionEngineConfiguration.NAME)
+public class KmsEncryptionEngineConfiguration implements EncryptionEngineConfiguration, AwsCredentialsConfig {
+    static final String S3_PREFIX = "s3://";
+    private static final String AWS_IAM = "iam";
+    private static final String AWS_IAM_ROLE = "role";
+    static final String NAME = "kms";
+
+    @JsonProperty("encryption_key")
+    private String encryptionKey;
+
+    @JsonProperty("encryption_key_directory")
+    private String encryptionKeyDirectory;
+
+    @JsonProperty("key_id")
+    private String keyId;
+
+    @JsonProperty("encryption_context")
+    private Map<String, String> encryptionContext;
+
+    @Valid
+    @Size(min = 1, message = "Region cannot be empty string")
+    @JsonProperty("region")
+    private String region;
+
+    @Valid
+    @Size(min = 20, max = 2048, message = "sts_role_arn length should be between 20 and 2048 characters")
+    @JsonProperty("sts_role_arn")
+    private String stsRoleArn;
+
+    @Valid
+    @JsonProperty("rotation_interval")
+    @DurationMin(hours = 2L, message = "Rotation interval must be at least 2 hours.")
+    private Duration rotationInterval = Duration.ofDays(30);
+
+    public String getEncryptionKey() {
+        return encryptionKey;
+    }
+
+    public String getEncryptionKeyDirectory() {
+        return encryptionKeyDirectory;
+    }
+
+    public String getKeyId() {
+        return keyId;
+    }
+
+    public Map<String, String> getEncryptionContext() {
+        return encryptionContext;
+    }
+
+    @Override
+    public String getRegion() {
+        return region;
+    }
+
+    @Override
+    public String getStsRoleArn() {
+        return stsRoleArn;
+    }
+
+    @Override
+    public AwsCredentialsOptions toCredentialsOptions() {
+        return AwsCredentialsOptions.builder()
+                .withRegion(region)
+                .withStsRoleArn(stsRoleArn)
+                .build();
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public boolean rotationEnabled() {
+        return isEncryptionKeyInS3();
+    }
+
+    @Override
+    public Duration getRotationInterval() {
+        return rotationInterval;
+    }
+
+    public boolean isEncryptionKeyInS3() {
+        return encryptionKeyDirectory != null && encryptionKeyDirectory.toLowerCase().startsWith(S3_PREFIX);
+    }
+
+    public KmsClient createKmsClient() {
+        final AwsContext awsContext = new AwsContext(this);
```

Comment:
Going along with my previous comments, these would end up using the `AwsContext` extension. You might need to move this code out of there though.